src/test/java/bio/terra/integration/AzureIntegrationTest.java

```diff
@@ -435,6 +444,66 @@ private void populateOmopTable() throws Exception {
     dataRepoFixtures.updateSettings(steward, releaseSnapshotId, "omop/settings.json");
   }
 
+  @Test
+  public void testSnapshotCreateFromRequest() throws Exception {
+    populateOmopTable();
+    UUID snapshotRequestId = makeSnapshotAccessRequest().getId();
+    SnapshotSummaryModel snapshotSummaryByRequest = makeSnapshotFromRequest(snapshotRequestId);
+    String columnName = "datarepo_row_id";
+    List<Object> personSnapshotRows =
+        dataRepoFixtures
+            .retrieveSnapshotPreviewById(
+                steward, snapshotSummaryByRequest.getId(), "person", 0, 100, null, columnName)
+            .getResult();
+    List<Object> conditionOccurrenceSnapshotRows =
+        dataRepoFixtures
+            .retrieveSnapshotPreviewById(
+                steward,
+                snapshotSummaryByRequest.getId(),
+                "condition_occurrence",
+                0,
+                100,
+                null,
+                columnName)
+            .getResult();
+    List<Object> conceptSnapshotRows =
+        dataRepoFixtures
+            .retrieveSnapshotPreviewById(
+                steward, snapshotSummaryByRequest.getId(), "concept", 0, 100, null, columnName)
+            .getResult();
+    assertThat(personSnapshotRows.size(), is(23));
+    // full table has 53 rows but only 49 map to existing person ids
+    assertThat(conditionOccurrenceSnapshotRows.size(), is(49));
+    // full table has 7 rows but only 5 are in the condition_occurrence table
+    assertThat(conceptSnapshotRows.size(), is(5));
+  }
+
+  private SnapshotAccessRequestResponse makeSnapshotAccessRequest() throws Exception {
+    String filename = "omop/snapshot-access-request.json";
+    SnapshotAccessRequestResponse accessRequest =
+        dataRepoFixtures.createSnapshotAccessRequest(steward, releaseSnapshotId, filename);
+    assertThat("Snapshot access request exists", accessRequest, notNullValue());
+    snapshotAccessRequestIds.add(accessRequest.getId());
+    return accessRequest;
+  }
+
+  private SnapshotSummaryModel makeSnapshotFromRequest(UUID requestSnapshotId) throws Exception {
+    SnapshotRequestModel requestSnapshot =
+        jsonLoader.loadObject(
+            "omop/snapshot-request-model-by-request-id.json", SnapshotRequestModel.class);
+    requestSnapshot.getContents().get(0).datasetName(datasetName);
+    requestSnapshot.getContents().get(0).getRequestIdSpec().snapshotRequestId(requestSnapshotId);
```

Comment:
To make it clear that these are setters, I would either write this as a fluent-style constructor, or use the `set` versions of the methods.
```suggestion
    requestSnapshot.getContents().get(0).setDatasetName(datasetName);
    requestSnapshot.getContents().get(0).getRequestIdSpec().setSnapshotRequestId(requestSnapshotId);
```